src/dpi.rs

```diff
@@ -1,80 +1,83 @@
+//! having an exact DPI value matters it should be safe to use `96 * scale_factor` on desktops and
+//! the web and `160 * scale_factor` on mobile platforms.
 //!
-//! Depending on the platform, the window's actual DPI factor may only be known after
-//! the event loop has started and your window has been drawn once. To properly handle these cases,
-//! the most robust way is to monitor the [`HiDpiFactorChanged`](../enum.WindowEvent.html#variant.HiDpiFactorChanged)
-//! event and dynamically adapt your drawing logic to follow the DPI factor.
+//! Winit's `Physical(Position|Size)` types correspond with the actual pixels on the device, and the
+//! `Logical(Position|Size)` types correspond to the physical pixels divided by the scale factor.
+//! All of Winit's functions return physical types, but can take either logical or physical
+//! coordinates as input, allowing you to use the most convenient coordinate system for your
+//! particular application.
 //!
-//! Here's an overview of what sort of DPI factors you can expect, and where they come from:
-//! - **Windows:** On Windows 8 and 10, per-monitor scaling is readily configured by users from the display settings.
-//! While users are free to select any option they want, they're only given a selection of "nice" DPI factors, i.e.
-//! 1.0, 1.25, 1.5... on Windows 7, the DPI factor is global and changing it requires logging out.
-//! - **macOS:** The buzzword is "retina displays", which have a DPI factor of 2.0. Otherwise, the DPI factor is 1.0.
-//! Intermediate DPI factors are never used, thus 1440p displays/etc. aren't properly supported. It's possible for any
-//! display to use that 2.0 DPI factor, given the use of the command line.
-//! - **X11:** On X11, we calcuate the DPI factor based on the millimeter dimensions provided by XRandR. This can
-//! result in a wide range of possible values, including some interesting ones like 1.0833333333333333. This can be
-//! overridden using the `WINIT_HIDPI_FACTOR` environment variable, though that's not recommended.
-//! - **Wayland:** On Wayland, DPI factors are set per-screen by the server, and are always integers (most often 1 or 2).
-//! - **iOS:** DPI factors are both constant and device-specific on iOS.
-//! - **Android:** This feature isn't yet implemented on Android, so the DPI factor will always be returned as 1.0.
-//! - **Web:** DPI factors are handled by the browser and will always be 1.0 for your application.
+//! Winit will dispatch a [`HiDpiFactorChanged`](crate::event::WindowEvent::HiDpiFactorChanged)
+//! event whenever a window's scale factor has changed. This gives you a chance to rescale your
+//! application's UI elements and adjust how the platform changes the window's size to reflect the
+//! new scale factor. If a window hasn't received a
+//! [`HiDpiFactorChanged`](crate::event::WindowEvent::HiDpiFactorChanged) event, then its
+//! scale factor is `1.0`.
```

Comment:
Should the event be generated unconditionally during startup so this assumption doesn't need to be encoded in every downstream application?

Reply:
Possibly, but that'll take implementation work on each platform and isn't implemented currently. Feel free to open a dedicated issue for it, though.